### PR TITLE
VM Instantiation successful! 

### PR DESCRIPTION
**Pull request by morpheyesh**

### CODE REVIEW

**Comment by indykish** [comment]
@morpheyesh Lets review the code when you are ready. Would like to see the test coverage as well and usage from an external entity as well (`megamd` . @rajthilakmca Review it as well.

File: compute/compute.go [hunk 1]
```diff
@@ -2,29 +2,27 @@ package compute
...
+
+	"github.com/megamsys/opennebula-go/api"
 	"github.com/megamsys/opennebula-go/flavor"
-"github.com/megamsys/opennebula-go/xmlUtil"
+	"github.com/megamsys/opennebula-go/xmlUtil"
 )
 
 type VirtualMachine struct {
-	OpenNebulaTemplateName   string
+	OpenNebulaTemplateName string
```
Comment (indykish) [inline compute/compute.go]: Why do you need the prefix `OpenNebula`
I see fields like `Bootstrap`, `RunList`, `Distro`, `SSHUser`, `SSHPort` will not be of any use

File: compute/compute.go [hunk 2]
```diff
@@ -45,42 +43,30 @@ flavorObj := flavor.FlavorOpts{TemplateName: VM.OpenNebulaTemplateName}
...
 
-/*
- * Updating templates
- */
-//data := `<VMTEMPLATE><ID>6</ID><UID>0</UID><GID>3</GID><UNAME>osradmin</UNAME><GNAME>NewUpdate</GNAME><NAME>supertest</NAME><PERMISSIONS><OWNER_U>1</OWNER_U><OWNER_M>1</OWNER_M><OWNER_A>0</OWNER_A><GROUP_U>0</GROUP_U><GROUP_M>0</GROUP_M><GROUP_A>0</GROUP_A><OTHER_U>0</OTHER_U><OTHER_M>0</OTHER_M><OTHER_A>0</OTHER_A></PERMISSIONS><REGTIME>1440059694</REGTIME><TEMPLATE><CONTEXT><NETWORK><![CDATA[YES]]></NETWORK><SSH_PUBLIC_KEY><![CDATA[$USER[SSH_PUBLIC_KEY]]]></SSH_PUBLIC_KEY></CONTEXT><CPU><![CDATA[1]]></CPU><CPU_COST><![CDATA[10]]></CPU_COST><DESCRIPTION><![CDATA[testtemplate...]]></DESCRIPTION><HYPERVISOR><![CDATA[kvm]]></HYPERVISOR><LOGO><![CDATA[images/logos/ubuntu.png]]></LOGO><MEMORY><![CDATA[512]]></MEMORY><MEMORY_COST><![CDATA[10]]></MEMORY_COST><SUNSTONE_CAPACITY_SELECT><![CDATA[YES]]></SUNSTONE_CAPACITY_SELECT><SUNSTONE_NETWORK_SELECT><![CDATA[YES]]></SUNSTONE_NETWORK_SELECT></TEMPLATE></VMTEMPLATE>`
+	template[0].Template.Cpu = VM.Cpu
+	template[0].Template.VCpu = VM.VCpu
+	template[0].Template.Memory = VM.Memory
 
+	finalXML := xmlUtil.VMTEMPLATE_POOL{}
```
Comment (indykish) [inline compute/compute.go]: please change the xmlUtil thing.

File: compute/compute_test.go [hunk 3]
```diff
@@ -8,8 +8,9 @@ import (
 
 func TestCreateVM(t *testing.T) {
 	Convey("CreateVM", t, func() {
```
Comment (indykish) [inline compute/compute_test.go]: Let us use one testing library - we have been using `gocheck` please use that.  I also going to add `prehooks` to reject code if it isn't "go fmt` ed.

File: xmlUtil/xmlUtil.go [hunk 4]
```diff
@@ -12,32 +10,59 @@ Rename this to UserTemplates (if this contains all the templates of an user)
 */
 type VMTEMPLATE_POOL struct {
```
Comment (indykish) [inline xmlUtil/xmlUtil.go]: The UserTemplates can be an array of UserTemplate[]

type UserTemplates []UserTemplate

**Comment by morpheyesh** [comment]
Yes, code clean up is pending and will do it tomorrow and push it.  Will be
ready before the builds.

**Pull request merged**